network/common/src/main/java/org/apache/spark/network/buffer/NioManagedBuffer.java

```diff
@@ -41,13 +41,13 @@ public long size() {
   }
 
   @Override
-  public ByteBuffer nioByteBuffer() throws IOException {
+  public LargeByteBuffer nioByteBuffer() throws IOException {
     return buf.duplicate();
   }
 
   @Override
   public InputStream createInputStream() throws IOException {
-    return new ByteBufInputStream(Unpooled.wrappedBuffer(buf));
+    return new ByteBufInputStream(Unpooled.wrappedBuffer(buf.firstByteBuffer()));
```

Comment:
we could wrap the entire `LargeByteBuffer` in a `LargeByteBufferInputStream` -- but I don't understand why there is already this level of indirection here so I thought I'd check.  This code is only for fetching shuffle blocks in any case, so other things will break if there is over 2gb in any case.